File: 896.单调数列.py
#
# @lc app=leetcode.cn id=896 lang=python3
#
# [896] 单调数列
#

# @lc code=start
class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        if len(A) == 1:
            return True
        result = [0]
        for i in range(len(A) - 1):
            temp = A[i] - A[i+1]
            result.append(temp)
        return (max(result) == 0 or min(result) == 0)
        
# @lc code=end

File: 896.单调数列.cpp
/*
 * @lc app=leetcode.cn id=896 lang=cpp
 *
 * [896] 单调数列
 */

// @lc code=start
class Solution {
public:
    bool isMonotonic(vector<int>& A) {
        vector<int> tmp1 = A;
        sort(A.begin(), A.end());
        if(A == tmp1)
            return true;
        else{
            sort(A.rbegin(), A.rend());
            if(A == tmp1)
                return true;
        }
        return false;
    }
};
// @lc code=end

